Makefile

```diff
@@ -1,11 +1,26 @@
-all: generate test lint
+SHELL := /usr/bin/env bash
+
+GOGC ?= 1000 # Reduce GC frequency during testing, default to 1000 if unset.
+
+all: generate test fuzz lint
 
-test: GOGC ?= 1000 # Reduce the GC frequency, default to 1000 if unset.
 test:
 	GOGC=$(GOGC) go test ./...
 .PHONY: test
 
+fuzz: FUZZTIME ?= 10s # The duration to run fuzz testing, default to 10s if unset.
+fuzz: # List all fuzz tests across the repo, and run them one at a time with the configured fuzztime.
+	@go run ./scripts/list_fuzz_tests | while read -r line; do \
+		package=$$(echo $$line | cut -d: -f1); \
+		func=$$(echo $$line | cut -d: -f2 | xargs); \
+		echo "Running ./$$package $$func for $(FUZZTIME)..."; \
+		GOGC=$(GOGC) go test ./$$package -fuzz=$$func -fuzztime=$(FUZZTIME); \
+		echo "Done."; \
+	done
```

Comment:
I did try this in my IDE. Some tests certainly run, and quickly. But I suspect it's quick because it didn't run most of the tests!

Reply:
It runs them once (a single seed). IMO, that's reasonable for unit tests and we can do longer runs in CI (e.g., in the merge queue?).

Reply:
That's not enough for development. Bugs have in the past come at iterations >3000 (due to the fairly naive search space exploration). I have no problem with tests taking >30s, but they have to have coverage.

Reply:
> It runs them once (a single seed). IMO, that's reasonable for unit tests and we can do longer runs in CI (e.g., in the merge queue?).

Ah, it doesn't. _We're_ running them once, `go test` runs them with a seed corpus (empty for us).

What we can do is to fill the "seed" corpus. E.g., by copying the generated corpus from `$(go env GOCACHE)/fuzz/github.com/filecoin-project/go-f3/test/ test/testdata/fuzz`. This corpus is selected by profiling code coverage.

When I did this (pushed as steb/seed-corpus) the tests took about 40s (compared to 56s on main) with identical coverage (for what that's worth)._This_ branch (with a single fuzz run) loses coverage of two cases:

https://github.com/filecoin-project/go-f3/blob/825e442519e71da18679e3333c68e1df735ec4a0/gpbft/gpbft.go#L692-L695
https://github.com/filecoin-project/go-f3/blob/825e442519e71da18679e3333c68e1df735ec4a0/gpbft/gpbft.go#L1138-L1140

I didn't test it with full fuzzing because merging fuzzing profiles is kind of a pain.

---

At this point, I'd go with fuzzing PLUS deterministic tests (where we only fuzz on-demand and in CI).

Reply:
Adding fuzzing (mainly CI or long-running) as an addition to deterministic tests similar to the current ones sounds great to me.

Reply:
Fuzz coprus (either explicit or in test data) will run with every tests.